analyses/fusion-summary/01-fusion-summary.R

```diff
@@ -0,0 +1,167 @@
+                   Gene1B %in% genes |
+                   Gene2B %in% genes)
+  } else if (!missing(fuses)) {
+    df <- filter(df, FusionName %in% fuses)
+  } else if (!missing(genes)) {
+    df <- filter(df,     
+                 Gene1A %in% genes |
+                   Gene2A %in% genes |
+                   Gene1B %in% genes |
+                   Gene2B %in% genes)
+  }
+  return(df)
+}
+
+#' Creates a TSV of the filtered fusion sheet
+#' @param df The filtered fusion data frame
+#' @param fuses List of explicit fusion names
+#' @param outputPath Path of the output file
+#' @return Writes a TSV to the output path
+generateOutput <- function(df, fuses, outputPath) {
+  # create a list to be used to reduce the table to its minimal set
+  # minimal set is the explicit fusions + any other non-specific fusions in the table
+  lvls = unique(c(fuses, sort(as.character(unique(df$FusionName)))))
+  df$FusionName = factor(df$FusionName, levels=lvls)
+  # convert the table to a table dropping the empty levels
+  tbl = table(droplevels(df)$Sample,df$FusionName)
+  # convert back to a matrix
+  mtx = as.data.frame.matrix(tbl)
+  # convert the rownames into the first row
+  mtx = setDT(mtx,keep.rownames=TRUE)
+  # rename the column of the rownames
+  colnames(mtx)[colnames(mtx) == 'rn'] <- 'Kids_First_Biospecimen_ID'
+  write.table(mtx, file=outputPath, sep="\t", quote=FALSE, row.names=FALSE)
+}
+
+#' Set up the options
+optionList <- list(
+  make_option(
+    opt_str = c("-d","--demographic_file"), type = "character",
+    default = NULL, help = "Path to the demographic file."
+  ),
+  make_option(
+    opt_str = c("-f","--fusions_file"), type = "character",
+    default = NULL, help = "Path to the fusions file."
+  ),
+  make_option(
+    opt_str = c("-o", "--output_dir"), type = "character",
+    default = "pbta-consolidated-fusions", help = "Output directory for output files."
+  )
+)
+
+#' Parse the options
+opt <- parse_args(OptionParser(option_list = optionList))
+
+#' Check the output directory
+if (!dir.exists(opt$output_dir)) {
+  dir.create(opt$output_dir)
+}
+
+#' Check that the files exist
+if (!file.exists(opt$demographic_file)) {
+  demo_missing <- paste("Error:", opt$demographic_file, "does not exist")
+  if (!file.exists(opt$fusions_file)) {
+    stop(paste(demo_missing, "\nError:", opt$fusions_file, "does not exist"))
+  } else
+    stop(demo_missing)
+} else if (!file.exists(opt$fusions_file)) {
+  stop(paste("Error:", opt$fusions_file, "does not exist"))
+}
+#' Load the files
+demo <- read.csv(opt$demographic_file, sep="\t")
+fuse <- read.csv(opt$fusions_file, sep="\t")
+
+#' Use the filters from above to get the biospecimen ids
+demoEpend.bios <- filter(demo,
+                        short_histology == ependSH)$Kids_First_Biospecimen_ID
+demoEmbry.bios <- filter(demo,
+                        broad_histology == embryBH &
+                          !(short_histology %in% embrySH))$Kids_First_Biospecimen_ID
+
```

Comment:
I think we can remove these steps now that you have the files that are not filtered. It is straightforward for an analyst to filter the files using `pbta-histologies.tsv` if they need to downstream.

Reply:
Thanks for the input. Removed.